File: Scripts/results/generated_code_results/RN_gpt/Social Networking/SocialSoftware/UserProfileScreen.js
import React from 'react';
import { View, Text, StyleSheet, Image, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import { useNavigation } from '@react-navigation/native';

const UserProfileScreen = () => {
  const navigation = useNavigation();

  return (
    <View style={styles.container}>
      {/* Header */}
      <View style={styles.header}>
        <TouchableOpacity>
          <Text style={styles.backText}>Voltar</Text>
        </TouchableOpacity>
        <View style={styles.headerIcons}>
          <TouchableOpacity>
            <Icon name="logo-instagram" size={24} color="white" />
          </TouchableOpacity>
          <TouchableOpacity>
            <Icon name="menu-outline" size={24} color="white" />
          </TouchableOpacity>
        </View>
      </View>

      {/* User Information */}
      <View style={styles.userInfo}>
        <Text style={styles.name}>Alisson Rochinski</Text>
        <View style={styles.iconRow}>
          <Icon name="at-outline" size={14} color="white" />
          <Text style={styles.handle}>arochinski</Text>
          <Text style={styles.website}>threads.net</Text>
        </View>
        <Text style={styles.bio}>UI Designer | Let's redesign the world</Text>
        <Text style={styles.email}>alisson.rochinski@gmail.com</Text>
        <View style={styles.followInfo}>
          <Image style={styles.profileImage} source={require('../assets/snack-icon.png')} />
          <Text style={styles.followers}>2957 seguidores</Text>
        </View>
        <TouchableOpacity style={styles.followButton}>
          <Text style={styles.followButtonText}>Follow</Text>
        </TouchableOpacity>
      </View>

      {/* Tabs */}
      <View style={styles.tabs}>
        <Text style={styles.tabText}>Chirps</Text>
        <Text style={styles.tabText}>Answers</Text>
      </View>

      {/* Post */}
      <View style={styles.post}>
        <View style={styles.postHeader}>
          <Image style={styles.postProfileImage} source={require('../assets/snack-icon.png')} />
          <Text style={styles.postHandle}>arochinski</Text>
          <Text style={styles.postTime}>2 min</Text>
        </View>
        <Text style={styles.postContent}>
          Tempor dolor proident duis incididunt et. Cillum incididunt irure nisi excepteur sit ad ipsum consectetur proident deserunt sint
        </Text>
        <View style={styles.postActions}>
          <TouchableOpacity>
            <Icon name="heart-outline" size={20} color="white" />
          </TouchableOpacity>
          <TouchableOpacity>
            <Icon name="chatbubble-outline" size={20} color="white" />
          </TouchableOpacity>
          <TouchableOpacity>
            <Icon name="share-outline" size={20} color="white" />
          </TouchableOpacity>
        </View>
        <Text style={styles.postStats}>7 answers · 59 likes</Text>
      </View>

      {/* Bottom Navigation */}
      <View style={styles.bottomNav}>
        <TouchableOpacity onPress={() => navigation.navigate('Search')}>
          <Icon name="search-outline" size={30} color="white" />
        </TouchableOpacity>
        <TouchableOpacity onPress={() => navigation.navigate('Activity')}>
          <Icon name="heart-outline" size={30} color="white" />
        </TouchableOpacity>
        <TouchableOpacity>
          <Icon name="add-circle-outline" size={30} color="white" />
        </TouchableOpacity>
        <TouchableOpacity>
          <Icon name="home-outline" size={30} color="white" />
        </TouchableOpacity>
        <TouchableOpacity onPress={() => navigation.navigate('Feed')}>
          <Icon name="person-outline" size={30} color="white" />
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#000',
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    padding: 10,
  },
  backText: {
    color: 'white',
    fontSize: 16,
  },
  headerIcons: {
    flexDirection: 'row',
    gap: 15,
  },
  userInfo: {
    padding: 20,
    alignItems: 'center',
  },
  name: {
    color: 'white',
    fontSize: 20,
    fontWeight: 'bold',
  },
  iconRow: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingVertical: 5,
  },
  handle: {
    color: 'white',
    fontSize: 14,
    marginLeft: 5,
  },
  website: {
    color: 'gray',
    fontSize: 14,
    marginLeft: 10,
  },
  bio: {
    color: 'white',
    textAlign: 'center',
    marginBottom: 5,
  },
  email: {
    color: 'white',
    marginBottom: 10,
  },
  followInfo: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 10,
  },
  profileImage: {
    width: 30,
    height: 30,
  },
  followers: {
    color: 'white',
    marginLeft: 10,
  },
  followButton: {
    backgroundColor: '#1D72F1',
    borderRadius: 5,
    paddingVertical: 5,
    paddingHorizontal: 20,
  },
  followButtonText: {
    color: 'white',
  },
  tabs: {
    flexDirection: 'row',
    justifyContent: 'center',
    borderBottomColor: 'gray',
    borderBottomWidth: 1,
    paddingVertical: 10,
  },
  tabText: {
    color: 'white',
    fontSize: 16,
    marginHorizontal: 20,
  },
  post: {
    padding: 20,
  },
  postHeader: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 10,
  },
  postProfileImage: {
    width: 40,
    height: 40,
  },
  postHandle: {
    color: 'white',
    fontSize: 16,
    marginLeft: 10,
  },
  postTime: {
    color: 'gray',
    fontSize: 14,
    marginLeft: 'auto',
  },
  postContent: {
    color: 'white',
  },
  postActions: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    paddingVertical: 10,
  },
  postStats: {
    color: 'gray',
    textAlign: 'center',
  },
  bottomNav: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
    height: 60,
    borderTopColor: 'gray',
    borderTopWidth: 1,
  },
});

export default UserProfileScreen;

File: Scripts/results/generated_code_results/Flutter/Social Networking/SocialSoftware/UserProfileScreen.dart
import 'package:flutter/material.dart';



class UserProfileScreen extends StatelessWidget {

  @override

  Widget build(BuildContext context) {

    return Scaffold(

      appBar: AppBar(

        leading: IconButton(

          icon: Icon(Icons.arrow_back),

          onPressed: () => Navigator.pop(context),

        ),

        title: Text('Voltar'),

        actions: [

          IconButton(

            icon: Icon(Icons.camera_alt),

            onPressed: () {

              // TODO: Implement camera functionality

            },

          ),

          IconButton(

            icon: Icon(Icons.menu),

            onPressed: () {

              // TODO: Implement menu functionality

            },

          ),

        ],

      ),

      body: SingleChildScrollView(

        child: Column(

          crossAxisAlignment: CrossAxisAlignment.start,

          children: [

            Padding(

              padding: const EdgeInsets.all(16.0),

              child: Row(

                crossAxisAlignment: CrossAxisAlignment.start,

                children: [

                  Stack(

                    children: [

                      CircleAvatar(

                        radius: 40,

                        backgroundImage: AssetImage('assets/images/template.png'),

                      ),

                      Positioned(

                        right: 0,

                        bottom: 0,

                        child: Icon(Icons.verified, color: Colors.blue),

                      ),

                    ],

                  ),

                  SizedBox(width: 16),

                  Expanded(

                    child: Column(

                      crossAxisAlignment: CrossAxisAlignment.start,

                      children: [

                        Text(

                          'Alisson Rochinski',

                          style: TextStyle(fontSize: 24, fontWeight: FontWeight.bold),

                        ),

                        Text(

                          'arochinski',

                          style: TextStyle(fontSize: 16, color: Colors.grey),

                        ),

                        SizedBox(height: 8),

                        Chip(

                          label: Text('threads.net'),

                          backgroundColor: Colors.grey[200],

                        ),

                        SizedBox(height: 8),

                        Text(

                          "UI Designer | Let's redesign the world\nalisson.rochinski@gmail.com",

                          style: TextStyle(fontSize: 16),

                        ),

                        SizedBox(height: 8),

                        Text(

                          '2957 seguidores',

                          style: TextStyle(fontSize: 16, color: Colors.grey),

                        ),

                      ],

                    ),

                  ),

                ],

              ),

            ),

            Padding(

              padding: const EdgeInsets.symmetric(horizontal: 16.0),

              child: OutlinedButton(

                child: Text('Follow'),

                style: OutlinedButton.styleFrom(

                  minimumSize: Size(double.infinity, 40),

                ),

                onPressed: () {

                  // TODO: Implement follow functionality

                },

              ),

            ),

            SizedBox(height: 16),

            DefaultTabController(

              length: 2,

              child: Column(

                children: [

                  TabBar(

                    tabs: [

                      Tab(text: 'Chirps'),

                      Tab(text: 'Answers'),

                    ],

                  ),

                  SizedBox(

                    height: 300, // Adjust this height as needed

                    child: TabBarView(

                      children: [

                        _buildChirpsList(),

                        Center(child: Text('No answers yet')),

                      ],

                    ),

                  ),

                ],

              ),

            ),

          ],

        ),

      ),

      bottomNavigationBar: BottomNavigationBar(

        type: BottomNavigationBarType.fixed,

        items: [

          BottomNavigationBarItem(icon: Icon(Icons.home), label: 'Home'),

          BottomNavigationBarItem(icon: Icon(Icons.search), label: 'Search'),

          BottomNavigationBarItem(icon: Icon(Icons.add_box), label: 'Create'),

          BottomNavigationBarItem(icon: Icon(Icons.favorite), label: 'Likes'),

          BottomNavigationBarItem(icon: Icon(Icons.person), label: 'Profile'),

        ],

        onTap: (index) {

          switch (index) {

            case 0:

              Navigator.pushNamed(context, 'Feed');

              break;

            case 1:

              Navigator.pushNamed(context, 'Search');

              break;

            case 3:

              Navigator.pushNamed(context, 'Activity');

              break;

          }

        },

      ),

    );

  }



  Widget _buildChirpsList() {

    return ListView(

      children: [

        _buildChirp(),

        // Add more chirps as needed

      ],

    );

  }



  Widget _buildChirp() {

    return Padding(

      padding: const EdgeInsets.all(16.0),

      child: Column(

        crossAxisAlignment: CrossAxisAlignment.start,

        children: [

          Row(

            children: [

              CircleAvatar(

                backgroundImage: AssetImage('assets/images/template.png'),

              ),

              SizedBox(width: 8),

              Text('arochinski', style: TextStyle(fontWeight: FontWeight.bold)),

              Spacer(),

              Text('2 min', style: TextStyle(color: Colors.grey)),

              IconButton(

                icon: Icon(Icons.more_horiz),

                onPressed: () {

                  // TODO: Implement more options functionality

                },

              ),

            ],

          ),

          SizedBox(height: 8),

          Text('Tempor dolor proident duis incididunt et. Cillum incididunt irure nisi excepteur sit ad ipsum consectetur proident deserunt sint'),

          SizedBox(height: 8),

          Row(

            children: [

              IconButton(

                icon: Icon(Icons.favorite_border),

                onPressed: () {

                  // TODO: Implement like functionality

                },

              ),

              Text('59'),

              IconButton(

                icon: Icon(Icons.chat_bubble_outline),

                onPressed: () {

                  // TODO: Implement comment functionality

                },

              ),

              Text('7'),

              IconButton(

                icon: Icon(Icons.repeat),

                onPressed: () {

                  // TODO: Implement repost functionality

                },

              ),

              IconButton(

                icon: Icon(Icons.send),

                onPressed: () {

                  // TODO: Implement send functionality

                },

              ),

            ],

          ),

        ],

      ),

    );

  }

}